### PR TITLE
Stat: recompute shared y range during streaming updates

### LINKED ISSUE #39425: Stat-Panel: Sparkline/Graph is not updated when streaming
**What happened**: A Stat-Panel does not refresh the graph, when using a streaming datasource. The shown value is refreshed as expected. The sparkline isn't.

**What you expected to happen**: Refresh the sparkline in sync with the shown value

**How to reproduce it (as minimally and precisely as possible)**: Use a streaming datasource like the mqtt example.

**Anything else we need to know?**:

**Environment**:
- Grafana version: v8.1.0
- Data source type & version: custom Datasource 
- OS Grafana is installed on: Windows
- User OS & Browser: Win + Chrome
- Grafana plugins: none (expect the custom Datasource)
- Others:

### PR DESCRIPTION
**Pull request by leeoniya**
Fixes #39425.

during streaming we normally skip running fieldOverrides (for perf), which is where the shared `field.state.range` is set. this copies the same (tweaked) code to the stat panel.

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]
LGTM
Peek 2021-09-21 09-57

(laggy is just the capture software :man_shrugging: )

File: public/app/plugins/panel/stat/StatPanel.tsx [hunk 1]
```diff
@@ -83,6 +87,28 @@ export class StatPanel extends PureComponent<PanelProps<StatPanelOptions>> {
...
+          if (!globalRange && (!isNumber(config.min) || !isNumber(config.max))) {
+            globalRange = findNumericFieldMinMax(data.series);
+          }
+          const min = config.min ?? globalRange!.min;
+          const max = config.max ?? globalRange!.max;
+          field.state = field.state ?? {};
+          field.state.range = { min, max, delta: max! - min! };
+        }
+      }
+    }
```
Comment (torkelo) [inline public/app/plugins/panel/stat/StatPanel.tsx:110]: Given that this logic is needed in almost any streaming graph panel, is there a way we can move this to a shared function.

It feels so hacky to have this concern (and amount of code) exist in each panel.

**Comment by torkelo** [comment]
Why does not updating the y-range in streaming make the sparkline x axis static?

File: public/app/plugins/panel/stat/StatPanel.tsx (hunk 1 above)
Comment (leeoniya) [reply public/app/plugins/panel/stat/StatPanel.tsx:110]: yes, i'll pull this out into a util fn

**Comment by leeoniya** [comment]
> Why does not updating the y-range in streaming make the sparkline x axis static?

check the console; it errors due to missing `field.state.range` and just stops updating, so i think followup redraw calls never reach there.

**Comment by torkelo** [comment]
> check the console; it errors due to missing field.state.range and just stops updating, so i think followup setData/redraw calls never reach there. i havent looked into exactly why it doesnt error on every future update though...didn't seem terribly important.

ah, that makes sense.

**Review by dprokop — APPROVED** [review]

**Pull request merged**